readthedocs/telemetry/collectors.py

```diff
@@ -0,0 +1,261 @@
+                "all": all_apt_packages,
+            },
+        }
+        return data
+
+    def _get_all_conda_packages(self):
+        """
+        Get all the packages installed by the user using conda.
+
+        This includes top level and transitive dependencies.
+        The output of ``conda list`` is in the form of::
+
+            [
+                {
+                    "base_url": "https://conda.anaconda.org/conda-forge",
+                    "build_number": 0,
+                    "build_string": "py_0",
+                    "channel": "conda-forge",
+                    "dist_name": "alabaster-0.7.12-py_0",
+                    "name": "alabaster",
+                    "platform": "noarch",
+                    "version": "0.7.12"
+                },
+                {
+                    "base_url": "https://conda.anaconda.org/conda-forge",
+                    "build_number": 0,
+                    "build_string": "pyh9f0ad1d_0",
+                    "channel": "conda-forge",
+                    "dist_name": "asn1crypto-1.4.0-pyh9f0ad1d_0",
+                    "name": "asn1crypto",
+                    "platform": "noarch",
+                    "version": "1.4.0"
+                }
+            ]
+        """
+        code, stdout, _ = self.run(
+            "conda", "list", "--json", "--name", self.version.slug
+        )
+        if code == 0 and stdout:
+            packages = self._safe_json_loads(stdout, [])
+            packages = [
+                {
+                    "name": package["name"],
+                    "channel": package["channel"],
+                    "version": package["version"],
+                }
+                for package in packages
+            ]
+            return packages
+        return []
+
+    def _get_pip_packages(self, include_all=False):
+        """
+        Get all the packages installed by the user using pip.
+
+        This includes top level and transitive dependencies.
+        The output of ``pip list`` is in the form of::
+
+            [
+                {
+                    "name": "requests-mock",
+                    "version": "1.8.0"
+                },
+                {
+                    "name": "requests-toolbelt",
+                    "version": "0.9.1"
+                },
+                {
+                    "name": "rstcheck",
+                    "version": "3.3.1"
+                },
+                {
+                    "name": "selectolax",
+                    "version": "0.2.10"
+                },
+                {
+                    "name": "slumber",
+                    "version": "0.7.1"
+                }
+            ]
+        """
+        cmd = ["python", "-m", "pip", "list", "--pre", "--local", "--format", "json"]
+        if not include_all:
+            cmd.append("--not-required")
+        code, stdout, _ = self.run(*cmd)
+        if code == 0 and stdout:
+            return self._safe_json_loads(stdout, [])
+        return []
```

Comment:
It would be good to use `None` for `NULL` values to be saved in the db.

Reply:
Not sure if that would be safe here, when processing that information we will have to deal with lists and None, don't think the sql would be more complex because of this. We are also already returning empty lists in other parts, like when the user doesn't specify apt packages in the config file.